source/extensions/filters/http/jwt_authn/authenticator.h

```diff
@@ -9,9 +9,39 @@ namespace Extensions {
 namespace HttpFilters {
 namespace JwtAuthn {
 
+class JwtMatcher;
+typedef std::shared_ptr<const JwtMatcher> JwtMatcherConstPtr;
+
+class RequirementVerifier;
+typedef std::shared_ptr<const RequirementVerifier> JwtVerifierPtr;
+
 class Authenticator;
 typedef std::unique_ptr<Authenticator> AuthenticatorPtr;
 
+// Verifies JWT based on the requirement rules.
+class RequirementVerifier {
+public:
+  virtual ~RequirementVerifier() {}
+  virtual ::google::jwt_verify::Status verify(std::vector<std::string> issuers) const PURE;
+};
+
+/**
+ *  Wraps RequirementRule configuration, which match request path based on values in
+ *  said configuration and returns matching RequirementVerifier.
+ */
+
+class JwtMatcher {
+public:
+  virtual ~JwtMatcher() {}
+
+  // Returns true if path matches requirement configuration, false otherwise.
+  virtual bool matches(const Http::HeaderString& path) const PURE;
+  // Returns the JwtVerifier pointer, or nullptr if not set.
+  virtual const JwtVerifierPtr getRequirementVerifier() const PURE;
+  // If JwtVerifierPtr in this matcher is AllowMissingOrFailedVerifier.
```

Comment:
can we move Matcher into a set of .h, and .cc file, and add some unit-test for them.
Same for Verifier.